resources/inputs/login-form.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { useTranslations } from "next-intl";
+import { Provider } from "@supabase/supabase-js";
+import { yupResolver } from "@hookform/resolvers/yup";
+import AuthService from "@/services/auth/auth-supabase.service";
+import form_style from "@/styles/forms/form.module.css";
+import auth_schema from "@/resources/inputs/form-schemas/auth-schema";
+import messageHandler from "@/core/message-handler";
+import { getStaticPropsWithTranslations } from "@/modules/lang/props";
+import { GetStaticProps } from "next";
+
+export const getStaticProps: GetStaticProps = getStaticPropsWithTranslations;
+
+interface LoginFormProps {
+  onLoginSuccess: (user: any, token: string) => void;
+}
+
+const LoginForm: React.FC<LoginFormProps> = ({ onLoginSuccess }) => {
+  const t = useTranslations("");
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(auth_schema),
+  });
+
+  const onSubmit = async (data: any) => {
+    try {
+      const response = await AuthService.loginUser(data.email, data.password);
+      if (response) {
+        messageHandler.handleSuccess(t("Login successful"));
+        onLoginSuccess(response.user, response.token as string);
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        messageHandler.handleError(t(error.message));
+      } else {
+        messageHandler.handleError(t("An unexpected error occurred"));
+      }
+    }
+  };
+
+  const handleOAuthClick = async (provider: Provider) => {
+    try {
+      const response = await AuthService.oauth(provider);
+      if (response && response.url) window.location.href = response.url;
+    } catch (error) {
+      if (error instanceof Error) {
+        messageHandler.handleError(t(error.message));
+      } else {
+        messageHandler.handleError(
+          t("An unexpected error occurred during OAuth login"),
+        );
+      }
+    }
+  };
+
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className={form_style.formContainer}
+    >
+      <div className={form_style.formInput}>
+        <input
+          {...register("email", { required: "Email is required" })}
+          type="email"
+          placeholder={t("Email")}
+          className={`${form_style.formControl} ${form_style.inputEmailIcon}`}
+        />
+        {errors.email && (
+          <span className={form_style.errorMessage}>
+            {t(errors.email.message)}
+          </span>
+        )}
+      </div>
```

Comment:
I dont want to see any Input on the components, they go in the resource folders.

Otherwise the can not be rehused and the platform will be a frankestein.